### PR TITLE
Topic/add sbi extension probe

### PR DESCRIPTION
**Pull request by atulkharerivos**
This is to fix https://github.com/rivosinc/salus/issues/67.

1) The first commit adds APIs for the SBI base interface
2) The second commit adds SBI extension probes to Tellus and cleans-up lint issues
3) The final commit adds SBI extension probes to the GuestVm along with lint and formatting fixes.

### CODE REVIEW

File: sbi/src/api/base.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+/// Returns the version of this SBI implementation.
+pub fn get_implementation_version() -> Result<u64> {
+    let msg = SbiMessage::Base(GetImplementationVersion);
+    // Safety: This ecall doesn't touch memory
+    unsafe { ecall_send(&msg) }
+}
+
+/// Checks if the given SBI extension is supported.
+pub fn probe_sbi_extension(sbi_extension_id: u64) -> Result<u64> {
```
Comment (abrestic-rivos) [inline sbi/src/api/base.rs:29]: Should this return a bool instead?

File: sbi/src/api/base.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+/// Returns the vendor that produced this machine(`mvendorid`).
+pub fn get_machine_vendor_id() -> Result<u64> {
+    let msg = SbiMessage::Base(GetMachineVendorID);
+    // Safety: This ecall doesn't touch memory
+    unsafe { ecall_send(&msg) }
+}
+
+/// Returns the architecture implementation ID this machine(`marchid`).
```
Comment (dgreid) [inline sbi/src/api/base.rs:42]: nit: ID of this machine

File: test-workloads/src/bin/tellus.rs [hunk 3]
```diff
@@ -109,7 +113,7 @@ fn exercise_pmu_functionality() {
     let start_flags = PmuCounterStartFlags::default();
     let result = start_counters(counter_index, 0x1, start_flags, 0);
     if result.is_err() && !matches!(result.err().unwrap(), sbi::Error::AlreadyStarted) {
-        result.expect("start_counters failed with result {result:?}");
```
Comment (abrestic-rivos) [inline test-workloads/src/bin/tellus.rs:112]: This condition can be simplified. Also `expect()` will print the error, so you can leave it as a `panic!` or drop the printing of `result`.

```
if !matches!(result, Ok(_) | Err(sbi::Error::AlreadyStarted)) {
    result.expect("start_counters failed");
}
```

**Review by abrestic-rivos — APPROVED** [review]
LGTM

**Review by dgreid — APPROVED** [review]
thanks!

**Pull request merged**